### ISSUE TITLE
PSR12 incorrect spaces autofix

### ISSUE THREAD

**Issue by Aliance** [squizlabs/PHP_CodeSniffer#2820]
```php
<?php

declare(strict_types=1);

namespace Test;

class MyTestClass
{
    public function myTestMethod():? \stdClass
    {
        $whereFinal[] = 'x.corr_amount <= ' .- ((int) $this->corrFrom);

        return random_int(0, 1) ? null : new stdClass();
    }
}
```

There are two errors: wrong spaces in return type hinting and missed space before `-` sign.
But then I run phpcs, It shows unexpected behavior:

`docker run --rm -v $(pwd)/test.php:/scripts/test.php texthtml/phpcs:3.4.2 phpcs --standard=PSR12 -n -p -s /scripts/test.php`

```
E 1 / 1 (100%)

FILE: /scripts/test.php
----------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
  9 | ERROR | [x] There must not be a space between the question mark
    |       |     and the type in nullable type declarations
    |       |     (PSR12.Functions.NullableTypeDeclaration.WhitespaceFound)
 11 | ERROR | [x] Expected at least 1 space after "."; 0 found
    |       |     (PSR12.Operators.OperatorSpacing.NoSpaceAfter)
 11 | ERROR | [x] Expected at least 1 space before "-"; 0 found
    |       |     (PSR12.Operators.OperatorSpacing.NoSpaceBefore)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 143ms; Memory: 6MB
```

Also, the phpcbf fix them with an error in both cases:

`docker run --rm -v $(pwd)/test.php:/scripts/test.php texthtml/phpcs:3.4.2 phpcbf --standard=PSR12 /scripts/test.php`

The result:
```php
<?php

declare(strict_types=1);

namespace Test;

class MyTestClass
{
    public function myTestMethod():?\stdClass
    {
        $whereFinal[] = 'x.corr_amount <= ' .  - ((int) $this->corrFrom);

        return random_int(0, 1) ? null : new stdClass();
    }
}
```

**Comment by gsherwood** [comment]
The errors that I get are:
```
  9 | ERROR | [x] There must be a single space between the colon and type in a return type declaration
  9 | ERROR | [x] There must not be a space between the question mark and the type in nullable type declarations
 11 | ERROR | [x] Expected at least 1 space after "."; 0 found
 11 | ERROR | [x] Expected at least 1 space before "-"; 0 found
```
and the code is fixed like this:
```php
<?php

declare(strict_types=1);

namespace Test;

class MyTestClass
{
    public function myTestMethod(): ?\stdClass
    {
        $whereFinal[] = 'x.corr_amount <= ' .  - ((int) $this->corrFrom);

        return random_int(0, 1) ? null : new stdClass();
    }
}
```
Are you running the latest version?

**Comment by Aliance** [comment]
I am running docker image from [texthtml/phpcs](https://hub.docker.com/r/texthtml/phpcs) with tag `3.4.2`:
```
$ docker run --rm -v $(pwd)/test.php:/scripts/test.php texthtml/phpcs:3.4.2 phpcs --version

PHP_CodeSniffer version 3.4.2 (stable) by Squiz (http://www.squiz.net)
```

**Comment by gsherwood** [comment]
The PSR12 standard was not released as stable until version 3.5.0, so you will need to upgrade to 3.5 if you want to accurately check your code against PSR12. Moving to 3.5.3 is recommended given the bug fixes that were released for PSR12 in those versions.
